pandas/core/internals/array_manager.py

```diff
@@ -983,14 +983,10 @@ def reduce(self: T, func: Callable) -> T:
             # TODO NaT doesn't preserve dtype, so we need to ensure to create
             # a timedelta result array if original was timedelta
             # what if datetime results in timedelta? (eg std)
-            if res is NaT and is_timedelta64_ns_dtype(arr.dtype):
-                result_arrays.append(np.array(["NaT"], dtype="timedelta64[ns]"))
-            else:
-                # error: Argument 1 to "append" of "list" has incompatible type
-                # "ExtensionArray"; expected "ndarray"
-                result_arrays.append(
-                    sanitize_array([res], None)  # type: ignore[arg-type]
-                )
+            dtype = arr.dtype if res is NaT else None
```

Comment:
havent looked too closely, but could this mess up on "std" with dt64?

Reply:
Is the following sufficient?

```
df = pd.DataFrame({'a': ['2022-01-01', '2022-01-02', pd.NaT, '2022-01-03']})
df['a'] = pd.to_datetime(df['a'])
print(df.std())
# a   1 days
# dtype: timedelta64[ns]
```

I don't think any tests hit _reduce with std and dt64. Can add.

Reply:
i think the relevant test would have skipna=False (or all-NaT values)

Reply:
Thanks - test added.

Reply:
test_std_datetime64_with_nat passes with the change you made here?  i may need more caffeine

Reply:
I see it now - thanks. I get the same behavior on main - that std should return a td64 but when the result is NaT and the input is dt64 we get dt64 instead. From what I can tell std i the only op that does dt64 -> td64 and the PR is doing no harm here. Okay to xfail?

Reply:
Sure, thanks for taking a look